app/src/main/java/at/bitfire/icsdroid/ui/screen/EditCalendarScreen.kt

```diff
@@ -0,0 +1,220 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.Delete
+import androidx.compose.material.icons.filled.Share
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import at.bitfire.icsdroid.R
+import at.bitfire.icsdroid.db.entity.Subscription
+import at.bitfire.icsdroid.model.CredentialsModel
+import at.bitfire.icsdroid.model.EditCalendarModel
+import at.bitfire.icsdroid.model.EditSubscriptionModel
+import at.bitfire.icsdroid.model.SubscriptionSettingsModel
+import at.bitfire.icsdroid.ui.partials.ExtendedTopAppBar
+import at.bitfire.icsdroid.ui.partials.GenericAlertDialog
+import at.bitfire.icsdroid.ui.theme.AppTheme
+import at.bitfire.icsdroid.ui.views.LoginCredentialsComposable
+import at.bitfire.icsdroid.ui.views.SubscriptionSettingsComposable
+
+@Composable
+fun EditCalendarScreen(
+    application: Application,
+    subscriptionId: Long,
+    onShare: (subscription: Subscription) -> Unit,
+    onExit: () -> Unit = {}
+) {
+    val credentialsModel: CredentialsModel = viewModel()
+    val subscriptionSettingsModel: SubscriptionSettingsModel = viewModel()
+    val editSubscriptionModel: EditSubscriptionModel = viewModel {
+        EditSubscriptionModel(application, subscriptionId)
+    }
+    val editCalendarModel: EditCalendarModel = viewModel {
+        EditCalendarModel(editSubscriptionModel, subscriptionSettingsModel, credentialsModel)
+    }
+    EditCalendarScreen(
+        inputValid = editCalendarModel.inputValid,
+        modelsDirty = editCalendarModel.modelsDirty,
+        successMessage = editCalendarModel.editSubscriptionModel.uiState.successMessage,
+        onDelete = editSubscriptionModel::removeSubscription,
+        onSave = {
+            editSubscriptionModel.updateSubscription(subscriptionSettingsModel, credentialsModel)
+        },
+        {
+            editSubscriptionModel.subscriptionWithCredential.value?.let {
+                onShare(it.subscription)
+            }
+        },
+        onExit,
```

Comment:
The same here, if you find it suitable, since we have named arguments for all callbacks, maybe is more readable to have `onExit = onExit`. Even though it may be a bit redundant. What do you think?

Reply:
Sure, I think I just forgot them :)